homeassistant/components/wittiot/sensor.py

```diff
@@ -0,0 +1,465 @@
+"""Platform for sensor integration."""
+import dataclasses
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONCENTRATION_PARTS_PER_MILLION,
+    CONF_HOST,
+    DEGREE,
+    PERCENTAGE,
+    UnitOfIrradiance,
+    UnitOfLength,
+    UnitOfPrecipitationDepth,
+    UnitOfPressure,
+    UnitOfSpeed,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+# from homeassistant.helpers.entity import EntityDescription
+from .const import DEVICE_NAME, DOMAIN, MAIN_DATA
+from .coordinator import WittiotDataUpdateCoordinator
+
+TYPE_TEMPINF = "tempinf"
+TYPE_HUMIDITYIN = "humidityin"
+TYPE_REL = "baromrelin"
+TYPE_ABS = "baromabsin"
+TYPE_TEMPOUT = "tempf"
+TYPE_HUMIOUT = "humidity"
+TYPE_WDIR = "winddir"
+TYPE_WS = "windspeedmph"
+TYPE_WG = "windgustmph"
+TYPE_SR = "solarradiation"
+TYPE_UV = "uv"
+TYPE_DWM = "daywindmax"
+TYPE_FEELLIKE = "feellike"
+TYPE_DEWP = "dewpoint"
+TYPE_RR = "rainratein"
+TYPE_ER = "eventrainin"
+TYPE_DR = "dailyrainin"
+TYPE_WR = "weeklyrainin"
+TYPE_MR = "monthlyrainin"
+TYPE_YR = "yearlyrainin"
+TYPE_PIEZO_RR = "rrain_piezo"
+TYPE_PIEZO_ER = "erain_piezo"
+TYPE_PIEZO_DR = "drain_piezo"
+TYPE_PIEZO_WR = "wrain_piezo"
+TYPE_PIEZO_MR = "mrain_piezo"
+TYPE_PIEZO_YR = "yrain_piezo"
+TYPE_PM25CH1 = "pm25_ch1"
+TYPE_PM25CH2 = "pm25_ch2"
+TYPE_PM25CH3 = "pm25_ch3"
+TYPE_PM25CH4 = "pm25_ch4"
+TYPE_PM25RTAQICH1 = "pm25_aqi_ch1"
+TYPE_PM25RTAQICH2 = "pm25_aqi_ch2"
+TYPE_PM25RTAQICH3 = "pm25_aqi_ch3"
+TYPE_PM25RTAQICH4 = "pm25_aqi_ch4"
+TYPE_PM2524HAQICH1 = "pm25_avg_24h_ch1"
+TYPE_PM2524HAQICH2 = "pm25_avg_24h_ch2"
+TYPE_PM2524HAQICH3 = "pm25_avg_24h_ch3"
+TYPE_PM2524HAQICH4 = "pm25_avg_24h_ch4"
+TYPE_CO2IN = "co2in"
+TYPE_CO224HIN = "co2in_24h"
+TYPE_CO2OUT = "co2"
+TYPE_CO224HOUT = "co2_24h"
+TYPE_CO2PM25 = "pm25_co2"
+TYPE_CO224HPM25 = "pm25_24h_co2"
+TYPE_CO2PM10 = "pm10_co2"
+TYPE_CO224HPM10 = "pm10_24h_co2"
+TYPE_CO2RTPM10 = "pm10_aqi_co2"
+TYPE_CO2RTPM25 = "pm25_aqi_co2"
+TYPE_CO2TEMP = "tf_co2"
+TYPE_CO2HUMI = "humi_co2"
+TYPE_LIGHTNING = "lightning"
+TYPE_LIGHTNINGTIME = "lightning_time"
+TYPE_LIGHTNINGNUM = "lightning_num"
```

Comment:
Do we still need all of these? You're only using them once currently, but I will start about translation keys in a moment and in these cases it might be relevant to keep them.